rflx/model/message.py

```diff
@@ -640,6 +641,62 @@ def __compute_field_condition(self, final: Field) -> expr.Expr:
         )
 
 
+class ParallelProofs:
+    def __init__(self) -> None:
+        self.proofs: List[
+            List[Tuple[expr.Expr, List[expr.Expr], expr.ProofResult, RecordFluxError, bool, bool]]
+        ] = []
+        self.current: List[
+            Tuple[expr.Expr, List[expr.Expr], expr.ProofResult, RecordFluxError, bool, bool]
+        ] = []
+
+    def add(
+        self,
+        goal: expr.Expr,
+        facts: List[expr.Expr],
+        expected: expr.ProofResult,
+        message: RecordFluxError,
+        negate: bool = False,
+        add_unsat: bool = False,
+    ) -> None:
+        # pylint: disable=too-many-arguments
+        self.current.append((goal, facts, expected, message, negate, add_unsat))
+
+    def push(self) -> None:
+        if self.current:
+            self.proofs.append(copy(self.current))
+            self.current.clear()
```

Comment:
What is the reason for not parallelizing all proofs?

Reply:
Both, `__prove_reachability` and `__prove_contraditions` have some non-trivial control flow which doesn't fit the other cases (and thus the ParallelProofs class) very well. Maybe they could be adapted, but I don't think its worth the effort right now.

Reply:
I see.